src/java.base/macosx/classes/sun/nio/fs/MacOSXWatchService.java

```diff
@@ -0,0 +1,636 @@
+                return MODIFY;
+            } else if (StandardWatchEventKinds.ENTRY_DELETE == watchEventKind) {
+                return DELETE;
+            } else if (StandardWatchEventKinds.OVERFLOW == watchEventKind) {
+                return OVERFLOW;
+            } else {
+                throw new UnsupportedOperationException(watchEventKind.name());
+            }
+        }
+
+        public static EnumSet<FSEventKind> setOf(final WatchEvent.Kind<?>[] events) {
+            final EnumSet<FSEventKind> eventSet = EnumSet.noneOf(FSEventKind.class);
+            for (final WatchEvent.Kind<?> event: events) {
+                if (event == null) {
+                    throw new NullPointerException("An element in event set is 'null'");
+                } else if (event == StandardWatchEventKinds.OVERFLOW) {
+                    continue;
+                }
+
+                eventSet.add(FSEventKind.of(event));
+            }
+
+            if (eventSet.isEmpty())
+                throw new IllegalArgumentException("No events to register");
+
+            return eventSet;
+        }
+    }
+
+    private enum WatchModifier {
+        SENSITIVITY_HIGH, SENSITIVITY_MEDIUM, SENSITIVITY_LOW;
+
+        public static WatchModifier of(final WatchEvent.Modifier watchEventModifier) {
+            if (ExtendedOptions.SENSITIVITY_HIGH.matches(watchEventModifier)) {
+                return SENSITIVITY_HIGH;
+            } if (ExtendedOptions.SENSITIVITY_MEDIUM.matches(watchEventModifier)) {
+                return SENSITIVITY_MEDIUM;
+            } if (ExtendedOptions.SENSITIVITY_LOW.matches(watchEventModifier)) {
+                return SENSITIVITY_LOW;
+            } else {
+                throw new UnsupportedOperationException(watchEventModifier.name());
+            }
+        }
+
+        public static EnumSet<WatchModifier> setOf(final WatchEvent.Modifier[] modifiers) {
+            final EnumSet<WatchModifier> modifierSet = EnumSet.noneOf(WatchModifier.class);
+            for (final WatchEvent.Modifier modifier : modifiers) {
+                if (modifier == null)
+                    throw new NullPointerException("An element in modifier set is 'null'");
+
+                modifierSet.add(WatchModifier.of(modifier));
+            }
+
+            return modifierSet;
+        }
+
+        public static double sensitivityOf(final EnumSet<WatchModifier> modifiers) {
+            if (modifiers.contains(SENSITIVITY_HIGH)) {
+                return 0.1;
+            } else if (modifiers.contains(SENSITIVITY_LOW)) {
+                return 1;
+            } else {
+                return 0.5; // aka SENSITIVITY_MEDIUM
+            }
+        }
+    }
+
+    private static class MacOSXWatchKey extends AbstractWatchKey {
+        private static final Unsafe unsafe = Unsafe.getUnsafe();
+
+        private static final long kFSEventStreamEventFlagRootChanged     = 0x00000020;
+
+        private static final Path relativeRootPath = theFS.getPath("");
```

Comment:
Same comment as for lines 51-52. Unsure of line 311 however as that refers to a specific constant. Should that be perhaps in `UnixConstants`?

Reply:
I'll change the spelling of all but `kFSEventStreamEventFlagRootChanged`, which I'd rather keep as is. The constant being spelled exactly the same as in the corresponding C header file makes it easier to check its correctness (https://developer.apple.com/documentation/coreservices/1455361-fseventstreameventflags/kfseventstreameventflagrootchanged?language=objc).
As for moving it to `UnixConstants`, I'm also not so sure as it is hardly a "Unix" constant, more like a macOS constant.

Reply:
How about I make `kFSEventStreamEventFlagRootChanged` into a local variable? It's only used in one place and doesn't have to be a static field really.

Reply:
Sounds good.